src/network/protocol.rs

```diff
@@ -697,11 +707,36 @@ impl<T: Config> UdpProtocol<T> {
         self.round_trip_time = millis - body.pong;
     }
 
+    /// Upon recveiving a `ChecksumReport`, add it to the checksum history
+    fn on_checksum_report(&mut self, body: &ChecksumReport) {
+        if self.last_added_checksum_frame < body.frame {
+            self.last_added_checksum_frame = body.frame;
+            self.checksum_history
+                .push_front((body.frame, body.checksum));
+        }
+
+        while self.checksum_history.len() > MAX_CHECKSUM_HISTORY_SIZE {
```

Comment:
If you pop before adding, you can avoid potentially growing and moving the VecDeque